map/src/infoblock/components/graph/RoadAttributesGraph.jsx

```diff
@@ -77,84 +99,111 @@ export default function RoadAttributesGraph({ name, data, width }) {
     }
 
+    function prepareDistance(value) {
+        if (value?.distance) {
+            if (value.distance > 0.01) {
+                return `${(value.distance / 1000).toFixed(2)} km`;
+            } else {
+                return '< 10 m';
+            }
+        }
+        return '0 m';
+    }
+
     function onMouseMoveGraph(e, chartRef) {
         if (!chartRef) {
             return;
         }
-        if (ctx.mapMarkerListener && ctx.selectedGpxFile && chartRef.current._active?.length > 0) {
+        if (ctx.selectedGpxFile && chartRef.current._active?.length > 0) {
             let selected = chartRef.current._active[0];
             if (selected) {
                 let pointList = TracksManager.getTrackPoints(ctx.selectedGpxFile);
                 const ind = data.datasets[selected.datasetIndex].index;
                 if (ind) {
-                    const lat = Object.values(pointList)[ind].lat;
-                    const lng = Object.values(pointList)[ind].lng;
-                    ctx.mapMarkerListener(lat, lng);
+                    const range = [ind, ind + Number(pointList[ind].segment.ext.length)];
+                    ctx.setTrackRange(range);
                 } else {
-                    ctx.mapMarkerListener(null);
+                    hideSelectedPointSegment();
                 }
             }
         } else {
-            ctx.mapMarkerListener(null);
+            hideSelectedPointSegment();
         }
     }
 
+    function hideSelectedPointSegment() {
+        ctx.setTrackRange([]);
+    }
+
     return (
         <Box sx={{ p: 0, width: Number(width.replace('px', '')) - 40 }}>
             <Divider sx={{ mt: '15px', mb: '12px' }} />
-            <Typography variant="inherit" sx={{ color: '#666666', fontWeight: 'bold', mb: 2 }}>
-                {name}
-            </Typography>
-            <Box sx={{ p: 0, width: Number(width.replace('px', '')) - 40, height: 100 }}>
+            <Grid container spacing={2}>
+                <Grid item xs={9}>
+                    <Typography variant="inherit" sx={{ color: '#666666', fontWeight: 'bold', mb: 2 }}>
+                        {name}
+                    </Typography>
+                </Grid>
+                <Grid item xs={3}>
+                    <Button size="small" sx={{ color: '#f8931d', mt: '-1px' }} onClick={() => setOpen(!open)}>
+                        Details
+                        {open ? <ExpandLess /> : <ExpandMore />}
+                    </Button>
+                </Grid>
+            </Grid>
+            <Box sx={{ p: 0, width: Number(width.replace('px', '')) - 42, height: 60 }}>
                 <Bar
                     margin={{ top: 0, right: 0, left: -20, bottom: 0 }}
                     style={{ fontSize: 10 }}
                     data={graphData}
                     options={options}
                     onMouseMove={(e) => onMouseMoveGraph(e, chartRef)}
+                    onMouseLeave={() => hideSelectedPointSegment()}
```

Comment:
onMouseLeave={hideSelectedPointSegment}

No need to create new function () => xxx() every time when you not use arguments such as: (e) => xxx(e)